From b0cd66f94611f42190393fa76f9f6aceb640bb8a Mon Sep 17 00:00:00 2001
From: hyphop <email2tema@gmail.com>
Date: Mon, 28 Oct 2019 09:11:17 +0300
Subject: [PATCH] dts kreskue vim1 vim2

---

diff --git a/arch/arm64/boot/dts/amlogic/Makefile b/arch/arm64/boot/dts/amlogic/Makefile
index 6d4c2d1..f89ba01 100644
--- a/arch/arm64/boot/dts/amlogic/Makefile
+++ b/arch/arm64/boot/dts/amlogic/Makefile
@@ -39,3 +39,6 @@ dtb-$(CONFIG_ARCH_MESON) += meson-sm1-khadas-vim3l.dtb
 dtb-$(CONFIG_ARCH_MESON) += krescue-vim3-a311d.dtb
 dtb-$(CONFIG_ARCH_MESON) += krescue-vim3-s922x.dtb
 dtb-$(CONFIG_ARCH_MESON) += krescue-vim3-s905d3.dtb
+dtb-$(CONFIG_ARCH_MESON) += krescue-vim1.dtb
+dtb-$(CONFIG_ARCH_MESON) += krescue-vim2.dtb
+
-- 
2.14.1

