From 3216c0fbaf1cf38534c44e9e6405334356d814d3 Mon Sep 17 00:00:00 2001
From: hyphop <email2tema@gmail.com>
Date: Tue, 3 Sep 2019 12:29:16 +0300
Subject: [PATCH] .gitignore local fix

---
 .gitignore | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/.gitignore b/.gitignore
index 2030c7a..aa665d2 100644
--- a/.gitignore
+++ b/.gitignore
@@ -145,3 +145,7 @@ x509.genkey
 
 # Clang's compilation database file
 /compile_commands.json
+arch/arm64/configs/kvims_defconfig
+/defconfig
+/modules
+
-- 
2.14.1

